addons/mail/static/src/core/common/mail_core_common_service.js

```diff
@@ -86,29 +78,8 @@ export class MailCoreCommon {
                 }
             });
             this.busService.subscribe("mail.record/insert", (payload) => {
-                if (payload.Thread) {
-                    this.store.Thread.insert(payload.Thread);
-                }
-                if (payload.Persona) {
-                    const personas = Array.isArray(payload.Persona)
-                        ? payload.Persona
-                        : [payload.Persona];
-                    for (const persona of personas) {
-                        this.store.Persona.insert(persona);
-                    }
-                }
-                const { LinkPreview: linkPreviews } = payload;
-                if (linkPreviews) {
-                    for (const linkPreview of linkPreviews) {
-                        this.store.LinkPreview.insert(linkPreview);
-                    }
-                }
-                const { Message: messageData } = payload;
-                if (messageData) {
-                    this.store.Message.insert({
-                        ...messageData,
-                        body: messageData.body ? markup(messageData.body) : messageData.body,
-                    });
+                for (const Model in payload) {
```

Comment:
could use iterator for key, value directly

Reply:
```js
Object.entries(payload).forEach((Model, data) =>
    this.store[Model].insert(data, { html: true })
)
```
vs
```js
for (const Model in payload) {
    this.store[Model].insert(payload[Model], { html: true });
}
```
The latter is more idiomatic IMO